### ISSUE TITLE
Reading gro files with more than 2 characters before decimal point in velocity

### ISSUE THREAD

**Issue by mattwthompson** [mdtraj/mdtraj#864]
I recently ran into an error in which mdtraj fails to load a *.gro file with 3 characters before the decimal point of a velocity element. Specifically, the issues is a velocity element like -10.0, which leaves no space between it and the previous velocity. I have highlighted the problem and made a dummy file with the problem line:

    1FOO    foo 9328   3.064   2.394   6.775-10.3772 -0.2281  2.8391

located at https://gist.github.com/mattwthompson/8a5ffaf15002c41df535. The file that this came from was produced by a gromacs simulation, like a confout.gro. It can also be read by gromacs to a grompp command.

~~Christoph and I have tracked it to line 344 in~~

~~https://github.com/mdtraj/mdtraj/blob/master/mdtraj/formats/gro.py#L344~~

~~which fails because ```_isint(line[15:20])``` returns ```None``` in line 482~~

~~https://github.com/mdtraj/mdtraj/blob/master/mdtraj/formats/gro.py#L482~~

~~We suggest changing it to ```_isint(line[15:20].strip())```~~

**Comment by mpharrigan** [comment]
I'm confused: isn't `line[15:20]` 3.064?

**Comment by rmcgibbo** [comment]
```
In [33]: line = '1FOO    foo 9328   3.064   2.394   6.775-10.3772 -0.2281  2.8391'

In [34]: line[15:20]
Out[34]: '8   3'
```

**Comment by mpharrigan** [comment]
My counting by eye has failed me yet again. What field should that be picking up?

**Comment by mattwthompson** [comment]
Hm, that doesn't appear to make any sense. Ignore the suggestion, it seems unlreated to the issue later in the line.

Not sure if the spaces are showing up in that line, but it should be 
>>> line = '    1FOO    foo 9328   3.064   2.394   6.775-10.3772 -0.2281  2.8391'
>>> line[15:20]
' 9328'

**Comment by rmcgibbo** [comment]
I think the function `_is_gro_coord` is wrong. It shouldn't use `str.split`. It should use explicit offsets, since you can't depend on there being whitespace.

**Comment by swails** [comment]
@leeping implemented a rigorous GRO parser in ParmEd that supports arbitrary precision GRO files.

See https://github.com/ParmEd/ParmEd/blob/master/parmed/gromacs/gromacsgro.py

If I find some time, I'll port it over here.  I'm pretty sure that @leeping's code there mimics the exact logic that Gromacs itself uses.

**Comment by ctk3b** [comment]
@leeping's implementation in ParmEd looks like the way to go. 

Regarding the immediate issue, I get the following which I think is because the regex in `_isint()` doesn't allow for any whitespace:
```python
In [1]: import mdtraj as md

In [2]: t = md.load('v_error.gro')
> /Users/ctk3b/science/mdtraj/mdtraj/formats/gro.py(359)_read_frame()
    358                 import ipdb; ipdb.set_trace()
--> 359                 raise Exception("Unexpected line in .gro file: "+line)
    360 

ipdb> line
'    1FOO    foo 9328   3.064   2.394   6.775-10.3772 -0.2281  2.8391\n'
ipdb> len(line.split())
8
ipdb> line[15:20]
' 9328'
ipdb> _isint(line[15:20])
ipdb> _isint(line[15:20].strip())
<_sre.SRE_Match object; span=(0, 4), match='9328'>
```

Alternatively to using `strip()`, the regex could just be updated.

**Comment by swails** [comment]
This should be fixed by #866.  It works on the `v_error.gro` file in your gist.
